temba/contacts/models.py

```diff
@@ -2667,6 +2669,16 @@ def _row_to_spec(self, row: list[str]) -> dict:
                     spec["fields"] = {}
                 key = mapping["key"]
                 spec["fields"][key] = value
+            elif mapping["type"] == "groups":
+                groups_names = map(lambda x: x.strip(), value.split(","))
+                for group_name in groups_names:
+                    try:
+                        group = self.org.all_groups.get(name__exact=group_name)
+                        if group.group_type == ContactGroup.TYPE_USER_DEFINED:
+                            spec["groups"] = spec.get("groups", []) + [str(group.uuid)]
+                    except ContactGroup.DoesNotExist:
+                        group = ContactGroup.create_static(self.org, self.created_by, group_name)
+                        spec["groups"] = spec.get("groups", []) + [str(group.uuid)]
```

Comment:
@ihor-palii - Quick question: is Mailroom already prepared to run the adding contacts to groups process?

Reply:
Yes. I found out that the mailroom has a code to add groups by UUID. So I just added code on the RapidPro side to create a group if it does not exist and then pass the UUID of that group to the mailroom.

Reply:
Awesome @ihor-palii